pkg/retry/retry.go

```diff
@@ -77,3 +114,86 @@ func OnError(
 		}
 	}
 }
+
+// Call executes a function repeatedly, according to the retry strategy until
+// it succeeds or returns an `ErrNonRetryable`, which indicates that no more
+// retries should be attempted.
+//
+// If no retry strategy is provided, it defaults to an exponential backoff strategy
+// with 25 max retries, 500ms initial delay, and 30000ms max delay.
+//
+// Returns the result from the work function and any error that occurred if retries are exhausted.
+func Call[T any](
+	ctx context.Context,
+	work func() (T, error),
+	retryStrategy ...RetryStrategyFunc,
+) (T, error) {
+	var (
+		result T
+		err    error
+	)
+	if retryStrategy == nil {
+		retryStrategy = []RetryStrategyFunc{DefaultExponentialDelay}
```

Comment:
Why not use `GetStrategy(ctx)` here?

Reply:
Was planning on having `retryStrategy` optional to `Call` but I agree that it does not provide better DX compared to using `GetStrategy`.